### DISCUSSION TITLE
Obsoletion warning is not shown for obsolete overrides of non-obsolete base members

### DISCUSSION THREAD

**Discussion by jhinder** [dotnet/csharplang#8937] [Language Ideas]
Champion issue: https://github.com/dotnet/csharplang/issues/8938

**Version Used**: VS 16.0 Preview 2

**Steps to Reproduce**:

```c#
abstract class Base
{
    public abstract void M();
}

class Derived : Base
{
    [Obsolete]
    public override void M() => throw null;

    public void M2()
    {
        M();
    }
}
```

**Expected Behavior**: The call to `M` will result in a warning about using an obsolete member.

**Actual Behavior**: QuickInfo will show "[deprecated]" for `Derived.M()`, but no warning is emitted for the call.

This is important for `Span<T>`, which overrides `Equals` and `GetHashCode`. Both of those are guaranteed to throw, and the obsoletion attribute of these methods contains important information about using these methods (e.g. to [use `==` instead of `Equals`](https://docs.microsoft.com/en-gb/dotnet/api/system.span-1.equals?view=netcore-2.2#System_Span_1_Equals_System_Object_)).

**Comment by gafter** [comment]
According to the language specification, the invocation binds to the original method, not the override.  Since the original method is not obsolete, no warning is given.

**Comment by YoshiRulz** [comment]
This footgun is now at least documented correctly. But I would like to see the restriction lifted, per my earlier comment in the linked issue:
> My use-case is to indicate that the "general" `object.Equals(object?)` overload shouldn't be used because, to preserve commutativity, it only mimics the `IEquatable<TSelf>.Equals` overload, and not the additional `IEquatable<int>.Equals` overload—that is, `instance.Equals(1)` may resolve to the wrong overload and always return `false`.

It's also *already used* in the BCL: for [`Span<T>`](https://github.com/dotnet/runtime/blob/v8.0.0/src/libraries/System.Private.CoreLib/src/System/Span.cs#L187-L207)/[`ReadOnlySpan<T>`](https://github.com/dotnet/runtime/blob/v8.0.0/src/libraries/System.Private.CoreLib/src/System/ReadOnlySpan.cs#L181-L201), [`HashCode`](https://github.com/dotnet/runtime/blob/v8.0.0/src/libraries/System.Private.CoreLib/src/System/HashCode.cs#L483-L504), [`NonCryptographicHashAlgorithm`](https://github.com/dotnet/runtime/blob/v8.0.0/src/libraries/System.IO.Hashing/src/System/IO/Hashing/NonCryptographicHashAlgorithm.cs#L330-L344), and [this mock struct](https://github.com/dotnet/runtime/blob/v8.0.0/src/libraries/System.Memory/tests/TInt.cs#L42-L48) (all `Equals` and/or `GetHashCode`), and for [`PasswordDeriveBytes.GetBytes`](https://github.com/dotnet/runtime/blob/v8.0.0/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/PasswordDeriveBytes.cs#L112-L160). It looks like `VectorSpan<T>`/`ReadOnlyVectorSpan<T>` copied `Span<T>` and will have it too.
